src/locale/locale.tsx

```diff
@@ -41,4 +43,25 @@ export const useI18n = () => {
   return useContext(I18nContext)!!;
 };
 
-export {I18n};
+/**
+ * A wrapper for i18n which doesn't return new reference even i18n is updated
+ */
+export const useI18nRef = () => {
+  const i18n = useI18n();
+  const ref = useRef(i18n);
+
+  useEffect(() => {
+    ref.current = i18n;
+  }, [i18n]);
+
+  return useMemo<I18n>(() => {
+    return {
```

Comment:
These returns below are just a getter pattern so that we can always point to latest `ref.current`.